docs/hub/app-apis.md

```diff
@@ -34,122 +34,210 @@ The [js-textile](#libraries) library allows you to create and edit Buckets owned
+
+While we're at it, we'll also create a default identity for our user, using the `Libp2pCryptoIdentity` object. In practice, you might have your own identity provider, or you might want to use a hierarchical key/wallet or mnemonic phrase to help store a users keys for them. Whatever you decide, Textile's generic identity interface should be able to support it.
 
 ```typescript
-import { Context } from '@textile/textile'
-import { Client } from '@textile/threads-client'
+const ctx = new Context()
+const identity = await Libp2pCryptoIdentity.fromRandom() // Random identity
+```
 
-let ctx: Context = new Context()
+The next step is to authenticate the user with your User Key and Secret. This will allow the user to store threads and buckets using your developer resources on the Hub.
 
-// `USER_KEY` comes from `tt key create`
-ctx = ctx.withAPIKey(USER_KEY)
+```typescript
+// Update the context WITH the user key information
+await ctx.withUserKey({
+   key: process.env.USER_API_KEY,
+   secret: process.env.USER_API_SECRET,
+   type: 1, // User key type
+ })
+// This will also return a promise with your updated context:
 ```
 
-##### Create a db with the user identity
+### Setup ThreadDB
 
-Your users can be defined by any available identity provider or use any private-key based identity, such as Libp2p's identities.
+Now we will connect to our remote ThreadsDB client. This will allow us to connect to a remote ThreadsDB on the Textile Hub. This is an alternative to a local-first database where keys and data are stored _locally_ on the device (see local first [database example]() for details on this).
 
 ```typescript
-const identity = await Libp2pCryptoIdentity.fromRandom()
-
 const db = new Client(ctx)
-const tok = await db.getToken(identity)
-ctx = ctx.withToken(tok)
+// API calls will now include the credentials created above
 ```
 
-##### Create a user owned thread for their buckets
+With the ThreadsDB instance ready to connect to the remote database, it is time to generate a user token. This allows us (the developer) to allocate user-scoped resources without our remote database. The app user (defined by their Identity created above) needs an API token to perform database operations. The API will give you one based on ID plus your developer credentials. The token will be added to the existing db.context. The token can also manually be added to a context if one has already been generated for the user.
 
-Buckets run on Threads, so your user will need a ThreadDB setup for thier own Buckets.
+```typescript
+const token = await db.getToken(identity)
+// Or, update it manually from an existing token
+// db.context.withToken(token)
+```
+
+Note that the context is actually reusable in future sessions (except it will likely expire and need to be updated). It can base stored in a session or however your app stores app state (securely).
```

Comment:
feels a bit confusing here. since you just created a token but i haven't _seen_ a context since the previous section. as a reader too, if i jump around, where did this context come from or where is it now? probably just show that `db.context` exists....